testclient/mainwindow.cpp

```diff
@@ -243,15 +251,10 @@ void MainWindow::printPBClicked()
 
     case DISCONNECTED:
         QMessageBox::information(this, tr("Error"), tr("Not Connected To a Printer"));
-        if(!core->pluginLoaded()){
-            addLog(tr("No plugin loaded !"));
-            addLog(tr("Requesting Firmware..."));
-            core->requestFirmware();
-        }
         break;
```

Comment:
No if we have a serial Connection and the status is DISCONNECTED that is a bug that needs fixing..the purpose if the DISCONNECTED state is to inform us no serial device is open .